Point only OS 10.9+ users at official git DL

jiffyclub opened the pull request:
Everyone else should download a "snow-leopard" installer.

See also swcarpentry/site#931.

jiffyclub committed:
Point only OS 10.9+ users at official git DL

Everyone else should download a "snow-leopard" installer.

wking commented:
On Tue, Apr 14, 2015 at 04:22:59PM -0700, Matt Davis wrote:

> Everyone else should download a "snow-leopard" installer.

See also #198.

wking commented:
On Tue, Apr 14, 2015 at 04:22:59PM -0700, Matt Davis wrote:

> - Point only OS 10.9+ users at official git DL

So when I look at [1](with my Linux User-Agent), I see a “click here
to download manually” link pointing at [2](http://sourceforge.net/projects/git-osx-installer/files/git-2.3.5-intel-universal-snow-leopard.dmg/download?use_mirror=autoselect).  I don't know if there's
some fancy OS-detection logic being used here, but since everyone is
getting their OS X installers from the same SourceForge project, it
seems like the best solution here is proper OS-detection on
git-scm.com.  It looks like there is some basic platform detection
going on there [3](https://github.com/git/git-scm.com/blob/4d308ca10ee15d44ec3572a4c12cb7bd832f2e60/app/controllers/downloads_controller.rb#L25), but I'm not sure if it has access to the OS
version.  I'm also not sure where the snow-leopard bit I see comes
from, but maybe git-scm.com is just checking for whichever relase was
cut most recently [4](https://github.com/git/git-scm.com/blob/4d308ca10ee15d44ec3572a4c12cb7bd832f2e60/lib/tasks/downloads.rake#L72).  There doesn't seem to be an OS-version field
in their download schema anyway [5](https://github.com/git/git-scm.com/blob/4d308ca10ee15d44ec3572a4c12cb7bd832f2e60/db/schema.rb#L74).  Anyhow, I don't see offhand how
to fix git-scm.com, but it seems like they'd be interested in having
working links for folks on different OS X versions, so I'd consider
reaching out to them to see if we can update the central site, instead
of setting up a SWC-specific fix that asks learners to figure out
their OS X version on their own.  That said, doing something like this
is certainly a reasonable short-term improvement.

jiffyclub commented:
When I go to http://git-scm.com/download/mac on my Mac running 10.10 it's also pointing me at the snow-leopard download, which definitely isn't the right one. I might need to revise this PR to point both sets of people at sourceforge and tell them which thing to download.

jiffyclub committed:
Direct recent Mac OS verions to mavericks installer

git-scm.org doesn't seem to reliably point people at an installer
based on their OS version, rather it seems to use the most recently
built installer [1].
So it's probably best to explicitly send people to the git-osx-installer
downloads and tell them which one to grab.

[1]: https://github.com/swcarpentry/workshop-template/pull/204#issuecomment-93172064

wking commented:
On Tue, Apr 14, 2015 at 09:56:08PM -0700, Matt Davis wrote:

> When I go to http://git-scm.com/download/mac on my Mac running 10.10
> it's also pointing me at the snow-leopard download, which definitely
> isn't the right one. I might need to revise this PR to point both
> sets of people at sourceforge and tell them which thing to download.

I've spun off git/git-scm.com#532 in case they have ideas about using
browser detection to automatically point folks at the appropriate
installer for their OS X version.  Until then, I agree that it's best
to just point folks at the SourceForge repository.

jiffyclub commented:
Can this and swcarpentry/site#931 be merged? I'm not sure who needs to give approval. @r-gaia-cs?

fmichonneau commented:
Hi @jiffyclub, sorry I had missed your initial PR. It looks good to me. It seems that I lost the privilege to be able to merge PRs in this repo though.

gvwilson commented:
Happy to merge this one, but site#931 is an issue, not a PR (?).
G

On 2015-05-20 6:40 PM, Matt Davis wrote:

> Can this and swcarpentry/site#931 
> https://github.com/swcarpentry/site/issues/931 be merged? I'm not 
> sure who needs to give approval. @r-gaia-cs 
> https://github.com/r-gaia-cs?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/swcarpentry/workshop-template/pull/204#issuecomment-104063959.

## 

Dr. Greg Wilson    | gvwilson@software-carpentry.org
Software Carpentry | http://software-carpentry.org

jiffyclub commented:
@fmichonneau I've added you to the tool-maintainers team so you can merge stuff in this repo (and lesson-template).

@gvwilson Ah yeah forgot I didn't make a PR for the site because I wasn't sure why we have two sets of installation instructions.

gvwilson commented:
Gotcha - merged.

rgaiacs commented:
Sorry for the delay in reply the question in this thread:

> Can this and swcarpentry/site#931 be merged? I'm not sure who needs to give approval. @r-gaia-cs?

Except for things related to Mac OS X I'm happy to review the pull requests **because** I don't have a machine with OS X.
